runtime/shared_common/OSCachesysv.cpp

```diff
@@ -2686,7 +2688,7 @@ SH_OSCachesysv::getControlFilePerm(char *cacheDirName, char *filename, bool *isN
 	return rc;
 }
 
-#endif
+#endif /* !defined(WIN32) */
```

Comment:
At first I thought this was a stray `#endif` but it closes the `#if` on line 2274.
There are nested tests of `#if !defined(WIN32)` which are redundant and some dead code in `#else` blocks: can you clean that up too, please?